### PR TITLE
Allow query checkout for all users.

### PR DESCRIPTION
**Pull request by Air-t**
Allow for fetching checkouts or checkout if the channel is inactive with additional permission `HANDLE_PAYMENTS` for apps and staff users. 
Raises `PermissionDenied` error instead of returning `None` while the channel is inactive for the checkout or for checkouts query
Allow to query checkout if the user knows the ID of the checkout

Fixes: https://linear.app/saleor/issue/SHOPX-693/allow-fetching-all-checkouts-with-handle-payments-permission
Fixes: https://linear.app/saleor/issue/SHOPX-694/allow-fetching-checkout-if-id-is-known-even-if-checkout-was-assigned
Fixes: https://linear.app/saleor/issue/SHOPX-688/raise-an-error-in-checkout-query-when-the-permissions-are-missing-for

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/graphql/checkout/schema.py [hunk 1]
```diff
@@ -51,6 +55,7 @@ class CheckoutQueries(graphene.ObjectType):
             "query checkouts that belong to other users: "
             f"{CheckoutPermissions.MANAGE_CHECKOUTS.name}, "
             f"{AccountPermissions.IMPERSONATE_USER.name}. "
+            f"{PaymentPermissions.HANDLE_PAYMENTS.name}. "
```
Comment (korycins) [inline saleor/graphql/checkout/schema.py:58]: With your changes, the description is not valid anymore. You can fetch any checkout. Permissions are required only when you want to fetch the checkout from in-active channel.

File: saleor/graphql/checkout/resolvers.py [hunk 2]
```diff
@@ -34,30 +39,28 @@ def resolve_checkout(info, token, id):
...
-        if user and checkout.user == user:
-            return checkout
+        return checkout
 
-    # resolve checkout for staff user
+    # resolve checkout for staff or app
     requester = get_user_or_app_from_context(info.context)
 
     if not requester:
         return None
```
Comment (korycins) [inline saleor/graphql/checkout/resolvers.py:52]: Why `None`? You already changed the way in line 63/60 to raise PermissionDenied.

File: saleor/graphql/checkout/schema.py (hunk 1 above)
Comment (Air-t) [reply saleor/graphql/checkout/schema.py:58]: Yes, so should I split the description into two parts - for active and inactive channel or leave it as it has no permission at all?

File: saleor/graphql/checkout/resolvers.py (hunk 2 above)
Comment (Air-t) [reply saleor/graphql/checkout/resolvers.py:52]: should we raise the same error or more specyfic for a case where the requestor is not found?

Comment (korycins) [reply saleor/graphql/checkout/resolvers.py]: When requestor is not found, then it means that the request is anonymous, and try to fetch the details for the in-active channel (at least from what I understand 😅 )

File: saleor/graphql/checkout/schema.py (hunk 1 above)
Comment (korycins) [reply saleor/graphql/checkout/schema.py:58]: You require permissions for checkouts placed on in-active channels, so I would place this info in the description.

File: saleor/graphql/checkout/resolvers.py (hunk 2 above)
Comment (Air-t) [reply saleor/graphql/checkout/resolvers.py]: ok so throwing an PermissionDenied should be good option here

**Review by korycins — APPROVED** [review]

**Pull request merged**